src/objects.h

```diff
@@ -245,12 +245,14 @@ enum {
         T_BARRIER,
         T_SYNCVAR,
         // atomic lists and records, thread local records
+    START_ENUM_RANGE(FIRST_ATOMIC_TNUM),
         T_FIXALIST,
         T_ALIST,
         T_AREC,
         T_AREC_INNER,
         T_TLREC,
         T_TLREC_INNER,
+    END_ENUM_RANGE(LAST_ATOMIC_TNUM),
     END_ENUM_RANGE(LAST_SHARED_TNUM),
```

Comment:
I wonder if the "atomic" TNUM range should perhaps be adjacent to the "shared" range (instead of being nested in it)?

For that matter, I have some doubts about the "shared" range in general -- mostly because I know little to nothing about the TNUMs in it (other than those your new "atomic" range, that is)

Reply:
I similarly have no idea, I just left the things where they are :)

Reply:
Just to clarify: I totally agree with leaving this as it is *for this PR*, but in the future, we may want to revise this (e.g. some of the the "shared" TNUMs should be perhaps among the "constant" TNUMs (?!?))